iceoryx2-ffi/cxx/include/iox2/response_mut_uninit.hpp

```diff
@@ -50,66 +55,122 @@ class ResponseMutUninit {
+    friend class ActiveRequest;
 
-    auto assume_init() -> ResponseMut<Service, ResponsePayload, ResponseHeader>;
+    template <ServiceType S, typename ResponsePayloadT, typename ResponseHeaderT>
+    friend auto assume_init(ResponseMutUninit<S, ResponsePayloadT, ResponseHeaderT>&& self)
+        -> ResponseMut<S, ResponsePayloadT, ResponseHeaderT>;
+
+    explicit ResponseMutUninit() = default;
 
     ResponseMut<Service, ResponsePayload, ResponseHeader> m_response;
 };
 
 template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
 inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::header() const
-    -> ResponseHeaderRequestResponse& {
-    IOX_TODO();
+    -> ResponseHeaderRequestResponse {
+    return m_response.header();
 }
 
 template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
 template <typename T, typename>
 inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::user_header() const -> const T& {
-    IOX_TODO();
+    return m_response.user_header();
 }
 
 template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
 template <typename T, typename>
 inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::user_header_mut() -> T& {
-    IOX_TODO();
+    return m_response.user_header_mut();
+}
+
+template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
+template <typename T, typename>
+inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::payload() const -> const T& {
+    return m_response.payload();
+}
+
+template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
+template <typename T, typename>
+inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::payload() const
+    -> iox::ImmutableSlice<ValueType> {
+    return m_response.payload();
+}
+
+template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
+template <typename T, typename>
+inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::payload_mut() -> T& {
+    return m_response.payload_mut();
 }
 
 template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
-inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::payload() const -> const ResponsePayload& {
-    IOX_TODO();
+template <typename T, typename>
+inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::payload_mut() -> iox::MutableSlice<ValueType> {
+    return m_response.payload_mut();
 }
 
 template <ServiceType Service, typename ResponsePayload, typename ResponseHeader>
-inline auto ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::payload_mut() -> ResponsePayload& {
-    IOX_TODO();
+template <typename T, typename>
+inline void ResponseMutUninit<Service, ResponsePayload, ResponseHeader>::write_payload(ResponsePayload&& payload) {
+    new (&payload_mut()) ResponsePayload(std::forward<T>(payload));
```

Comment:
Can we do here perfect forwarding as well, like in the `RequestMutUninit::write_payload()` case.